spec.emu

```diff
@@ -4,20 +4,308 @@
 <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/8.4/styles/github.min.css">
 <script src="./spec.js"></script>
 <pre class="metadata">
-title: Proposal Title Goes Here
-stage: -1
-contributors: Your Name(s) Here
+title: String Dedent
+status: proposal
+stage: 1
+contributors: Justin Ridgewell
+location: https://tc39.es/proposal-string-dedent/
 </pre>
 
-<emu-clause id="sec-demo-clause">
-  <h1>This is an emu-clause</h1>
-  <p>This is an algorithm:</p>
-  <emu-alg>
-    1. Let _proposal_ be *undefined*.
-    1. If IsAccepted(_proposal_),
-      1. Let _stage_ be *0*.
-    1. Else,
-      1. Let _stage_ be *-1*.
-    1. Return ? ToString(_proposal_).
-  </emu-alg>
+<emu-clause id="sec-scope">
+  <h1>Scope</h1>
+  <p>
+    This is the spec text of the <a href="https://github.com/tc39/proposal-string-dedent/">String Dedent proposal</a> in ECMAScript.
+    String dedenting removes leading indentation from each line of a multiline string.
+  </p>
+</emu-clause>
+
+<emu-clause id="sec-properties-of-the-string-constructor">
+  <h1>Properties of the String Constructor <emu-xref href="#sec-properties-of-the-string-constructor"></emu-xref></h1>
+
+  <ins class="block">
+
+  <emu-clause id="sec-string.dedent">
+    <h1>String.dedent ( _templateOrFn_, ..._substitutions_ )</h1>
+    <emu-note>
+      <p>The `String.dedent` function may be called with either a template array and a variable number of substitutions, or it may be called with a function which itself takes a template array and a variable number of substitutions.</p>
+      <p>When called with a template and a variable number of substitutions, a string is returned which has common leading indentation removed from all lines with non-whitespace.</p>
+      <p>When called with a function, a closure is returned which wraps the function. When the closure is called with a template and variable number of substitutions, the common leading indentation of the temlpate is removed and the result of calling the function with this new dedented template and substitutions is returned.</p>
+      <p>Common leading indentation is defined as the longest sequence of whitespace characters that match exactly on every line that contains non-whitespace. Empty lines and lines which contain only whitespace do not affect common leading indentation.</p>
+      <p>When removing the common leading indentation, lines which contain only whitespace will have all whitespace removed.</p>
+    </emu-note>
+    <emu-alg>
+      1. If Type(_templateOrFn_) is Object and IsCallable(_templateOrFn_) is *true*, then
```

Comment:
Can the type not be object, while IsCallable is true?

Reply:
Actually, `IsCallable` checks type is `Object`. I'll remove.

Reply:
af8443f